lib/miq_automation_engine/service_models/miq_ae_service_generic_object.rb

```diff
@@ -28,7 +28,11 @@ def ae_user_identity
     def method_missing(method_name, *args)
       ae_user_identity unless @ae_user
       args = convert_params_to_ar_model(args)
-      results = object_send(method_name, *args)
+      results = if @object.property_method_defined?(method_name)
+                  @object.send(:_call_automate, method_name.to_s, *args)
+                else
+                  object_send(method_name, *args)
+                end
```

Comment:
While I like this I suggest that we move the logic to the ActiveRecord side to keep the internal GO logic there.  Maybe create a `go_send` or `generic_object_send` method and move this logic there.  We should try to keep the logic in the service model to the minimal we need to call AR object.

Reply:
Since generic_object is mainly used by automate, I would suggest the method in AR side to include not only `property_method`, but also `property_attribute` and `property_association`